src/ripple/rpc/handlers/LedgerHandler.cpp

```diff
@@ -134,27 +134,37 @@ doLedgerGrpc(RPC::GRPCContext<org::xrpl::rpc::v1::GetLedgerRequest>& context)
 
     if (request.transactions())
     {
-        for (auto& i : ledger->txs)
+        try
         {
-            assert(i.first);
-            if (request.expand())
+            for (auto& i : ledger->txs)
             {
-                auto txn =
-                    response.mutable_transactions_list()->add_transactions();
-                Serializer sTxn = i.first->getSerializer();
-                txn->set_transaction_blob(sTxn.data(), sTxn.getLength());
-                if (i.second)
+                assert(i.first);
+                if (request.expand())
                 {
-                    Serializer sMeta = i.second->getSerializer();
-                    txn->set_metadata_blob(sMeta.data(), sMeta.getLength());
+                    auto txn = response.mutable_transactions_list()
+                                   ->add_transactions();
+                    Serializer sTxn = i.first->getSerializer();
+                    txn->set_transaction_blob(sTxn.data(), sTxn.getLength());
+                    if (i.second)
+                    {
+                        Serializer sMeta = i.second->getSerializer();
+                        txn->set_metadata_blob(sMeta.data(), sMeta.getLength());
+                    }
```

Comment:
I _think_ the things that may `throw` here are the calls to `getSerializer()`.  If I got that wrong, then the rest of this comment may not apply.

I see two possible issues with the way this code is structured.

1. If one transaction in a ledger throws, then you will not return any transactions later in that ledger.

2. It looks like it may be possible to return an empty or partially filled entry in a `transactions_list` if an exception does occur.

I think these problems can be fixed by restructuring the code a bit.  Consider the following approach which moves the `try/catch`(s) inside of the `for` loop.
```
    if (request.transactions())
    {
        for (auto& i : ledger->txs)
        {
            assert(i.first);
            if (request.expand())
            {
                try
                {
                    Serializer const sTxn = i.first->getSerializer();
                    std::optional<Serializer> const sMeta(
                        i.second ? std::optional<Serializer>(
                                       i.second->getSerializer())
                                 : std::nullopt);

                    auto txn = response.mutable_transactions_list()
                                   ->add_transactions();

                    txn->set_transaction_blob(sTxn.data(), sTxn.getLength());
                    if (i.second)
                        txn->set_metadata_blob(
                            sMeta->data(), sMeta->getLength());
                }
                catch (std::exception const&)
                {
                    JLOG(context.j.error())
                        << __func__
                        << " - Error deserializing transaction in ledger "
                        << ledger->info().seq
                        << ". skipping transaction. "
                           "you should look into this further";
                }
            }
            else
            {
                try
                {
                    auto const& hash = i.first->getTransactionID();
                    response.mutable_hashes_list()->add_hashes(
                        hash.data(), hash.size());
                }
                catch (std::exception const&)
                {
                    JLOG(context.j.error())
                        << __func__
                        << " - Error deserializing transaction hash in ledger "
                        << ledger->info().seq
                        << ". skipping transaction. "
                           "you should look into this further";
                }
            }
        }
    }
```
With this restructure we do the work that might `throw` before we create a response entry.  I think this will help keep any response that is provided from being empty or partial.  We also attempt to return any transactions that might be in the ledger following the transaction with the serialization issue.